soroban-env-host/src/auth.rs

```diff
@@ -582,24 +734,37 @@ impl AuthorizationManager {
             return;
         }
         self.call_stack.pop();
-        for tracker in &mut self.trackers {
+        for tracker in &mut self.account_trackers {
             tracker.pop_frame();
         }
+
+        let mut i = 0;
+        while i < self.invoker_contract_trackers.len() {
+            self.invoker_contract_trackers[i].pop_frame();
```

Comment:
What's going on here? Why is this popping first?

Reply:
We pop frame from a tracker (similar to account trackers) and if it goes out of scope (== it's local call stack is empty) we simply remove it. This basically guarantees that contract tracker only affects a single 'top-level' function call.